federation/executor.go

```diff
@@ -0,0 +1,323 @@
+package federation
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/samsarahq/thunder/graphql"
+	"github.com/samsarahq/thunder/graphql/introspection"
+	"golang.org/x/sync/errgroup"
+)
+
+type ExecutorClient interface {
+	Execute(ctx context.Context, req *graphql.Query) ([]byte, error)
+}
+
+// Executor has a map of all the executor clients such that it can execute a 
+// subquery on any of the federated servers.
+// The planner allows it to coordinate the subqueries being sent to the federated servers
+type Executor struct {
+	Executors map[string]ExecutorClient
+	planner   *Planner  
+}
+
+func fetchSchema(ctx context.Context, e ExecutorClient) ([]byte, error) {
+	query, err := graphql.Parse(introspection.IntrospectionQuery, map[string]interface{}{})
+	if err != nil {
+		return nil, err
+	}
+
+	return e.Execute(ctx, query)
+}
+
+
+func NewExecutor(ctx context.Context, executors map[string]ExecutorClient) (*Executor, error) {
+	// Fetches the schemas from the executors clients
+	schemas := make(map[string]*introspectionQueryResult)
+	for server, client := range executors {
+		schema, err := fetchSchema(ctx, client)
+		if err != nil {
+			return nil, fmt.Errorf("fetching schema %s: %v", server, err)
```

Comment:
I'm wondering how you're thinking about partial failures here.  What happens if one of the introspection queries fails (maybe a temporary network error)?  Would the caller of `NewExecutor` be responsible for retrying?